Console/Command/NostoClearQueueCommand.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+namespace Nosto\Tagging\Console\Command;
+
+use Exception;
+use Magento\Framework\Amqp\Config;
+use Magento\Framework\App\ResourceConnection;
+use Nosto\NostoException;
+use RuntimeException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class NostoClearQueueCommand extends Command
+{
+    /**
+     * Nosto Queues
+     */
+    private const QUEUE_TOPICS = [
+        'nosto_product_sync.update',
+        'nosto_product_sync.delete'
+    ];
+
+    /**
+     * @var Config
+     */
+    private Config $amqpConfig;
+
+    /**
+     * @var ResourceConnection
+     */
+    private ResourceConnection $resourceConnection;
+
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        Config $amqpConfig
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->amqpConfig = $amqpConfig;
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        // Define command name.
+        $this->setName('nosto:clear:queue')
+            ->setDescription('Clear all message queues for Nosto product sync topics.');
+        parent::configure();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        try {
+            foreach (self::QUEUE_TOPICS as $topicName) {
+                $this->clearQueue($topicName, $io);
+            }
+
+            $io->success('Successfully cleared message queues.');
+            return 0;
+        } catch (NostoException $e) {
+            $io->error('An error occurred while clearing message queues: ' . $e->getMessage());
+            return 1;
+        }
+    }
+
+    /**
+     * Clear MySql and RabbitMq queues by name.
+     *
+     * @param string $topicName
+     * @param SymfonyStyle $io
+     * @return void
+     */
+    private function clearQueue(string $topicName, SymfonyStyle $io): void
+    {
+        $this->clearRabbitMQQueue($topicName, $io);
+        $this->clearDBQueues($topicName, $io);
```

Comment:
There's no need to clear both at the same time, you need to check which one is being used and clear only either RabbitMQ or MySQL